kv/kvserver: TestRejectedLeaseDoesntDictateClosedTimestamp failed

kv/kvserver.TestRejectedLeaseDoesntDictateClosedTimestamp [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9122901?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9122901?buildTab=artifacts#/) on master @ [0a72a49eb6e4ce5823b53d02125441c3da2e94f5](https://github.com/cockroachdb/cockroach/commits/0a72a49eb6e4ce5823b53d02125441c3da2e94f5):


```
=== RUN   TestRejectedLeaseDoesntDictateClosedTimestamp
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestRejectedLeaseDoesntDictateClosedTimestamp2323733602
    test_log_scope.go:79: use -show-logs to present logs inline
    replica_closedts_test.go:622: lease request unexpectedly finished. err: [NotLeaseHolderError] lease held by different store; r61: replica (n2,s2):2 not lease holder; current lease is repl=(n1,s1):1 seq=1 start=0,0 exp=1679126617.592028251,4 pro=1679126611.592028251,4
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestRejectedLeaseDoesntDictateClosedTimestamp2323733602
--- FAIL: TestRejectedLeaseDoesntDictateClosedTimestamp (5.95s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/replication
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRejectedLeaseDoesntDictateClosedTimestamp.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25607

kv/kvserver.TestRejectedLeaseDoesntDictateClosedTimestamp [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9246184?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9246184?buildTab=artifacts#/) on master @ [0fcc33bc2870961b9387999d8a9fed97fccbb2ae](https://github.com/cockroachdb/cockroach/commits/0fcc33bc2870961b9387999d8a9fed97fccbb2ae):


```
=== RUN   TestRejectedLeaseDoesntDictateClosedTimestamp
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestRejectedLeaseDoesntDictateClosedTimestamp2508951177
    test_log_scope.go:79: use -show-logs to present logs inline
    replica_closedts_test.go:622: lease request unexpectedly finished. err: [NotLeaseHolderError] lease held by different store; r63: replica (n2,s2):2 not lease holder; current lease is repl=(n1,s1):1 seq=1 start=0,0 exp=1679736261.099205612,4 pro=1679736255.099205612,4
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestRejectedLeaseDoesntDictateClosedTimestamp2508951177
--- FAIL: TestRejectedLeaseDoesntDictateClosedTimestamp (6.88s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRejectedLeaseDoesntDictateClosedTimestamp.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestRejectedLeaseDoesntDictateClosedTimestamp [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9256176?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9256176?buildTab=artifacts#/) on master @ [d107217dac5d817cc115bc0e97b7e53c0f2878bf](https://github.com/cockroachdb/cockroach/commits/d107217dac5d817cc115bc0e97b7e53c0f2878bf):


```
=== RUN   TestRejectedLeaseDoesntDictateClosedTimestamp
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestRejectedLeaseDoesntDictateClosedTimestamp408966624
    test_log_scope.go:79: use -show-logs to present logs inline
    replica_closedts_test.go:622: lease request unexpectedly finished. err: [NotLeaseHolderError] lease held by different store; r63: replica (n2,s2):2 not lease holder; current lease is repl=(n1,s1):1 seq=1 start=0,0 exp=1679817316.838444346,4 pro=1679817310.838444346,4
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestRejectedLeaseDoesntDictateClosedTimestamp408966624
--- FAIL: TestRejectedLeaseDoesntDictateClosedTimestamp (6.84s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRejectedLeaseDoesntDictateClosedTimestamp.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestRejectedLeaseDoesntDictateClosedTimestamp [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9264549?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9264549?buildTab=artifacts#/) on master @ [2bd2c806ab3044569b09e0a205b5bc0452ad4e2b](https://github.com/cockroachdb/cockroach/commits/2bd2c806ab3044569b09e0a205b5bc0452ad4e2b):


```
=== RUN   TestRejectedLeaseDoesntDictateClosedTimestamp
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestRejectedLeaseDoesntDictateClosedTimestamp2390422086
    test_log_scope.go:79: use -show-logs to present logs inline
    replica_closedts_test.go:622: lease request unexpectedly finished. err: [NotLeaseHolderError] lease held by different store; r63: replica (n2,s2):2 not lease holder; current lease is repl=(n1,s1):1 seq=1 start=0,0 exp=1679904417.924450354,4 pro=1679904411.924450354,4
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestRejectedLeaseDoesntDictateClosedTimestamp2390422086
--- FAIL: TestRejectedLeaseDoesntDictateClosedTimestamp (5.87s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRejectedLeaseDoesntDictateClosedTimestamp.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestRejectedLeaseDoesntDictateClosedTimestamp [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9308486?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9308486?buildTab=artifacts#/) on master @ [bb5c97b5d2073a27e7ec88654b05ac913112f610](https://github.com/cockroachdb/cockroach/commits/bb5c97b5d2073a27e7ec88654b05ac913112f610):


```
=== RUN   TestRejectedLeaseDoesntDictateClosedTimestamp
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestRejectedLeaseDoesntDictateClosedTimestamp1412720107
    test_log_scope.go:79: use -show-logs to present logs inline
    replica_closedts_test.go:622: lease request unexpectedly finished. err: [NotLeaseHolderError] lease held by different store; r63: replica (n2,s2):2 not lease holder; current lease is repl=(n1,s1):1 seq=1 start=0,0 exp=1680082582.116183688,4 pro=1680082576.116183688,4
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestRejectedLeaseDoesntDictateClosedTimestamp1412720107
--- FAIL: TestRejectedLeaseDoesntDictateClosedTimestamp (5.87s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99639 kv/kvserver: TestRejectedLeaseDoesntDictateClosedTimestamp failed [C-test-failure O-robot T-kv-replication branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRejectedLeaseDoesntDictateClosedTimestamp.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/client_raft_test.go
+++ b/pkg/kv/kvserver/client_raft_test.go
@@ -2093,10 +2093,10 @@ func runReplicateRestartAfterTruncation(t *testing.T, removeBeforeTruncateAndReA
 				},
 			},
 			RaftConfig: base.RaftConfig{
-				// Don't timeout raft leaders or range leases (see the relation between
-				// RaftElectionTimeoutTicks and RangeLeaseActiveDuration). This test expects
+				// Don't timeout raft leaders or range leases. This test expects
 				// tc.Servers[0] to hold the range lease for the range under test.
 				RaftElectionTimeoutTicks: 1000000,
+				RangeLeaseDuration:       time.Minute,
 			},
 		}
 	}

--- a/pkg/kv/kvserver/closed_timestamp_test.go
+++ b/pkg/kv/kvserver/closed_timestamp_test.go
@@ -680,12 +680,12 @@ func TestClosedTimestampFrozenAfterSubsumption(t *testing.T) {
 				ServerArgs: base.TestServerArgs{
 					Settings: cs,
 					RaftConfig: base.RaftConfig{
-						// We set the raft election timeout to a small duration. This should
-						// result in the node liveness duration being ~3.6 seconds. Note that
+						// We set the raft election timeout to a small duration. Note that
 						// if we set this too low, the test may flake due to the test
 						// cluster's nodes frequently missing their liveness heartbeats.
 						RaftHeartbeatIntervalTicks: 5,
 						RaftElectionTimeoutTicks:   6,
+						RangeLeaseDuration:         3 * time.Second,
 					},
 					Knobs: base.TestingKnobs{
 						Server: &server.TestingKnobs{

--- a/pkg/kv/kvserver/replica_closedts_test.go
+++ b/pkg/kv/kvserver/replica_closedts_test.go
@@ -539,7 +539,7 @@ func TestRejectedLeaseDoesntDictateClosedTimestamp(t *testing.T) {
 				RangeLeaseRenewalFraction: -1,
 				// Also make expiration-based leases last for a long time, as the test
 				// wants a valid lease after cluster start.
-				RaftElectionTimeoutTicks: 1000,
+				RangeLeaseDuration: time.Minute,
 			},
 			Knobs: base.TestingKnobs{
 				Server: &server.TestingKnobs{

